### ISSUE TITLE
Coadding with spline regression instead of sampling

### ISSUE THREAD

**Issue by mzechmeister** [mzechmeister/viper#26]
Currently

![grafik](https://user-images.githubusercontent.com/77435060/201129542-286ded54-2528-44ae-8f28-094f1baca478.png)
![grafik](https://user-images.githubusercontent.com/77435060/201124355-cd05ef76-23f9-4a11-8b8c-19563ec9147e.png)

Resampling:
https://github.com/mzechmeister/viper/blob/d7a72df5bec7d89e36f3072f145a01795283661f/viper.py#L536-L539
Coadding:
https://github.com/mzechmeister/viper/blob/d7a72df5bec7d89e36f3072f145a01795283661f/viper.py#L766-L768
Errors of spectra not used in coadding.

**Comment by JanaKoehler1985** [comment]
It is using the errors of the data as well as the telluric spectra itself for the weighting and coadding of the single spectra.
In the following line the weighting of the later coadding is created and saved:
` tpl_all[o,2][n] = S_tell * 1./(e/np.nanmean(f))   #**2 `

**Comment by mzechmeister** [comment]
So, currently the weighting goes like:

https://github.com/mzechmeister/viper/blob/d7a72df5bec7d89e36f3072f145a01795283661f/viper.py#L539
https://github.com/mzechmeister/viper/blob/d7a72df5bec7d89e36f3072f145a01795283661f/viper.py#L762
https://github.com/mzechmeister/viper/blob/d7a72df5bec7d89e36f3072f145a01795283661f/viper.py#L767

Meaning weighting is done with 1/e not 1/e^2. I see you considered this. Well, ideally it should be 1/e^2.
The exponent (or even a wealth of other converting function) for S_tell likely has potential for tedious fine tuning.

A further possibility to prevent overweighting of high S/N pixels (say 200 or 1000), which might be overestimated due to systematics, might be to add a jitter term 1/(e^2+(cont_flux/200)^2).

**Comment by JanaKoehler1985** [comment]
Exactly. I have made tests with 1/e as well as with 1/e² and for some reasons 1/e gave the better results. Nevertheless, I am aware this is maybe not the best solution. As said before, it is all work in progress. 

I  will try to do some tests with your suggestion and to see how it goes. What exactly stands cont_flux for?

**Comment by mzechmeister** [comment]
cont_flux: Ups, right it can be just flux. If we don't believe in S/N = flux/e_lim > 200, than we add in quadrature e_lim=f/200.
